### ISSUE TITLE
Prioritize ingredients from inventory over kitchen storage

### ISSUE THREAD

**Issue by BlayTheNinth** [blay09/CookingForBlockheads#422]
No need for a toggle or anything, the inventory should just always take priority.

Blocked by #673

Originally from CurseForge:
> I have a suggestion for this mod! I'll use an example to explain it. I have Pam's HarvestCraft, so let's say I'm trying to make a Hamburger. I need to make Toast, and I do already have Flour, Dough, Bread, and Butter in a cabinet, but I kind of want to make the Toast from scratch, both because I feel like it and so I won't waste ingredients for the next time when I may not feel like making it from scratch. So, I go to my Cooking Table to make Flour. But then, when I make Dough from it, instead of using the Flour I just made in my inventory, it uses the Flour that's in the cabinet. Same goes for when I make Heavy Cream for the Butter and use Dough to make Bread, then Bread and Butter to make Toast. Now I have a bunch of ingredients in my inventory that I have to dump back in the cabinet.
It's not that big of a deal, but it is rather annoying, especially if the ingredients are stored in different cabinets. So, I was thinking, maybe you could make it so that it prioritizes ingredients in your inventory, searching your inventory to find them before checking in the cabinets.

**Comment by Syndaryl** [comment]
That might be a configuration option - perhaps a toggle switch visible in the GUI for the kitchen crafting table!

Toast is actually a good example in another way, as the toaster allows you to make toast, and then you end up with the toast in your inventory. Everything you pull out of the stove of course also goes into your inventory. Lacking a button to automatically unpack my groceries (which I want in real life thank you very much) I'd prefer it to pull from my inventory by priority...

but I think the toggle switch is the best compromise

**Comment by alfalfascout** [comment]
I agree, having a bunch of ingredients dumped to my inventory is kind of a pain! I'd love to see an option to prefer ingredients from my inventory when cooking.

**Comment by BlayTheNinth** [comment]
Added in Minecraft 26.1
